flytekit/types/structured/structured_dataset.py

```diff
@@ -478,7 +478,7 @@ def to_literal(
                 python_val,
                 df_type,
                 protocol,
-                python_val.file_format,
+                sdt.format or typing.cast(StructuredDataset, python_val).DEFAULT_FILE_FORMAT,
```

Comment:
Will `sdt.format` ever be None?

Reply:
yes, It will. https://github.com/flyteorg/flytekit/blob/d1702fc3bf552c4c0725c879a586914c1f4e5d91/flytekit/models/types.py#L181